src/network/connection.js

```diff
@@ -155,8 +155,8 @@ module.exports = class Connection {
         })
 
         this.logError(error.message, { stack: e.stack })
-        await this.disconnect()
         this.rejectRequests(error)
+        await this.disconnect()
```

Comment:
This bug seems to happen right below this, inside `onTimeout` as well. I supposed that poses the same deadlock issue? @tulios @danielbatica

Reply:
Added #956